components/esp-tls/esp_tls.c

```diff
@@ -150,13 +163,11 @@ esp_err_t esp_tls_set_global_ca_store(const unsigned char *cacert_pem_buf, const
     if (global_cacert != NULL) {
         mbedtls_x509_crt_free(global_cacert);
     }
```

Comment:
@paulreimer Could you move this `global_cacert` check and the corresponding free inside the new `esp_tls_init_global_ca_store()` API at the beginning? And the remove the initial NULL check from there.

Reply:
Hi @ChiragAtal -- I can do this, but then repeated calls to `esp_tls_init_global_ca_store()` will wipe the store if it was already configured, e.g. you can't run it multiple times. Right now repeated calls to `esp_tls_set_global_ca_store` will wipe the store, but init is no-op if it is already allocated. Should I change it to the new behaviour?